src/common/evaluable.ts

```diff
@@ -60,16 +60,16 @@ export interface Evaluable {
    * Simplifies this Evaluable when possible.
    *
    * @param {Context} ctx context for the evaluation
-   * @param {false | string[]} alwaysEvaluate keys to be considered present even if they are not present in the context
-   * @param {false | string[]} deferEvaluate keys to be considered not present unless they are also in `alwaysEvaluate`;
-   *  when `alwaysEvaluate` is set to `false` and `deferEvaluate` is an array, every key that is not in `deferEvaluate`
+   * @param {false | string[]} strictKeys keys to be considered present even if they are not present in the context
+   * @param {false | string[]} optionalKeys keys to be considered not present unless they are also in `strictKeys`;
+   *  when `strictKeys` is set to `false` and `optionalKeys` is an array, every key that is not in `optionalKeys`
    *  is considered to be present and thus will be evaluated
    * @returns {Result | Evaluable} simplified value or itself
    */
   simplify(
     ctx: Context,
-    alwaysEvaluate?: false | string[],
-    deferEvaluate?: false | string[]
+    strictKeys?: false | string[],
+    optionalKeys?: false | string[]
```

Comment:
Probably should be reversed?

`extraKeys?: string[]` - new name for `ignoreKeys`

`strictKeys?: string[]`

Also, is `false |` necessary?  Given that the argument is optional?  So you can test whether it is `undefined` instead of `false`.

Reply:
I don't think I get it. In my opinion `strictKeys` instead os `ignoreKey` makes more sense as those are the keys that should be strictly evaluated no matter if they are in the context or not.

As for `optionalKeys` are those that could be evaluated if they are in the context or a part of the `strictKeys`.

What's in your mind?

Reply:
> I don't think I get it. In my opinion `strictKeys` instead os `ignoreKey` makes more sense as those are the keys that should be strictly evaluated no matter if they are in the context or not.
> 
> As for `optionalKeys` are those that could be evaluated if they are in the context or a part of the `strictKeys`.
> 
> What's in your mind?

Ah, yeah you're right.  That makes sense.  Only thing then:

> Also, is false | necessary? Given that the argument is optional? So you can test whether it is undefined instead of false.

Reply:
It's not, I'll remove the `false` and follow David's recommendation [here](https://github.com/briza-insurance/illogical/pull/210#discussion_r691188204)